components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/FederatedAuthenticatorConfig.java

```diff
@@ -101,9 +108,18 @@ public static FederatedAuthenticatorConfig build(OMElement federatedAuthenticato
                     Property[] propertiesArr = propertiesArrList.toArray(new Property[propertiesArrList.size()]);
                     federatedAuthenticatorConfig.setProperties(propertiesArr);
                 }
+            } else if ("DefinedBy".equals(elementName)) {
+                federatedAuthenticatorConfig.setDefinedByType(
+                        IdentityConstants.DefinedByType.valueOf(element.getText()));
             }
         }
 
+        // TODO: Remove warn log, once feature is ready.
+        if (federatedAuthenticatorConfig.getDefinedByType() == null) {
+            federatedAuthenticatorConfig.setDefinedByType(IdentityConstants.DefinedByType.SYSTEM);
+            LOG.warn("The defined by type is not set for the : " + federatedAuthenticatorConfig.getName());
```

Comment:
Let's have a debug log here rather than a warn. Also, let's indicate that 'system' is set by default

Reply:
Addressed